frontend/src/pages/UserAddPage.jsx

```diff
@@ -0,0 +1,194 @@
+import React, { useState, useEffect } from 'react';
+import { useNavigate } from 'react-router';
+import { useUser } from '../hooks/useUser';
+import { useApi } from '../hooks/useApi';
+import Spinny from '../components/Spinny';
+
+const UserAddPage = () => {
+  const navigate = useNavigate();
+  const { isLogged, isAdmin } = useUser();
+  const [username, setUsername] = useState('');
+  const [name, setName] = useState('');
+  const [password, setPassword] = useState('');
+  const [repeatPassword, setRepeatPassword] = useState('');
+  const [isNewAdmin, setIsNewAdmin] = useState(false);
+  const [message, setMessage] = useState('');
+  const [error, setError] = useState('');
+
+  const { data, isLoading, fetchFromApi } = useApi({ method: 'POST' });
+
+  useEffect(() => {
+    if (!isLogged) {
+      navigate('/login');
+      return;
+    }
+    if (!isAdmin) {
+      navigate('/');
+    }
+  }, [isLogged, isAdmin, navigate]);
+
+  useEffect(() => {
+    if (data) {
+      if (data.success) {
+        setMessage('User created successfully!');
+        setError('');
+        setUsername('');
+        setName('');
+        setPassword('');
+        setRepeatPassword('');
+        setIsNewAdmin(false);
+        // Redirect to users list after a 2s is enough time to show the message I guess?
+        setTimeout(() => navigate('/users'), 2000);
+      } else {
+        setError(data.message || 'Failed to create user.');
+        setMessage('');
+      }
+    }
+  }, [data, navigate]);
+
+  const handleSubmit = async e => {
+    e.preventDefault();
+    setError('');
+    setMessage('');
+
+    if (password !== repeatPassword) {
+      setError('Passwords do not match.');
+      return;
+    }
+
+    if (!username || !name || !password) {
+      setError('Please fill in all required fields.');
+      return;
+    }
+
+    const payload = {
+      username,
+      name,
+      password,
+      ...(isAdmin && { is_admin: isNewAdmin }),
+    };
+
+    await fetchFromApi('/users', payload);
```

Comment:
```suggestion
    const formData = new FormData();
    formData.append('username', username);
    formData.append('name', name);
    formData.append('password', password);
    formData.append('is_admin', isNewAdmin);

    await fetchFromApi('/users/add', { body: formData});
```
Changed url and type of send data.